class-wc-settings-page-komoju.php

```diff
@@ -365,10 +365,12 @@ private function fetch_all_payment_methods()
         try {
             $all_payment_methods = $api->paymentMethods();
             $methods_by_slug     = [];
+            $wc_currency         = get_woocommerce_currency();
 
             foreach ($all_payment_methods as $payment_method) {
-                $slug = $payment_method['type_slug'];
-                if (isset($methods_by_slug[$slug])) {
+                $slug        = $payment_method['type_slug'];
+                $pm_currency = $payment_method['currency'];
+                if (isset($methods_by_slug[$slug]) && $pm_currency !== $wc_currency) {
```

Comment:
This seems wrong to me.
This was checking `methods_by_slug` is already set, we are skipping the process.
Now it is checking `When this slug is already set AND the cureencies are different, then skip`
Which means, in the first check when this slug is not set yet, it will alwas skip this if statement and will set the slug.

Based on the above reason, this should check `||` or !(A && B)

Reply:
Good point. Actually I intentionally made this logic in this way.

Given that WC has `USD` as the default "default currency", it's possible that this part of code is called with the default currency is in non-`JPY`. Then, if we _always_ reject payment methods with mismatched currency (like `isset($methods_by_slug[$slug]) || $pm_currency !== $wc_currency`), it can end up with no payment method available, because PMs such as Konbini are available only in JPY.
Therefore I relaxed the condition. With the current logic, it will accept payment methods if 1) the currency matches, OR if 2) there is no alternative one available with the same `type_slug`. IOW, as negation, it rejects PMs that 1) have alternatives, AND 2) the currency does not match.

Yep indeed I first thought it was a bad idea, and eventually I found out that it would be better, seeing how [my first attempt here](https://github.com/degica/komoju-woocommerce/commit/54a7e96d9a7524af2638b1b683373ea579476c42) broke our tests.)

![KOMOJU for WooCommerce Admin -- lets me add and remove specialized payment gateways (failed) (attempt 2)](https://github.com/user-attachments/assets/a7ea1b77-0e69-4fec-9496-c9aa060ec2da)

That said I think that's not an intuitive logic, I'll add a comment into the code for future reference!

Reply:
Okay, I see. This is not about preventing registering the payment method but changing the priority.
In this case, I recommend to change the logic like below for the better redability.

    // If $slug is not set, register
    if (!isset($methods_by_slug[$slug])) {
        $methods_by_slug[$slug] = $payment_method;
    } else {
        // If $slug is already registered and
        // the payment currency matches the WooCommerce currency then override it
        if ($pm_currency === $wc_currency) {
            $methods_by_slug[$slug] = $payment_method;
        }
    }

Reply:
Good call, I've pushed another commit! 🙏

Reply:
Thanks! 👍